Bash/InstallHuntress-macOS-bash.sh

```diff
@@ -61,22 +61,172 @@ install_system_extension=false
+
+EOF
+}
+
+# Using logger function to provide helpful logs within RMM tools in addition to log file
 logger() {
     echo "$dd -- $*";
     echo "$dd -- $*" >> $log_file;
 }
 
-logger "Huntress install script last updated $scriptVersion"
+# This is for machines where the Huntress install is corrupted, damaged, or missing assets. 
+# This will try to fully remove Huntress before attempting a reinstall
+reregister() {
+    logger "Starting uninstall, please be aware this can take up to a minute"
+
+    countProcesses=$(ps aux | grep "HuntressAgent" -c)
+    if [ $countProcesses -le 1 ]; then
+        logger "Huntress process not running, System Extension may hang for a minute on uninstall"
+    fi
+    
+    # skip trying to stop the services and file deletes if no assets are found to avoid clogging log with errors
+    installCheck=$(checkForAssets)
+    if [ $installCheck == 1 ]; then
+        uninstall_system_extension
+    
+        cd "/Library/Application Support/Huntress/HuntressAgent"
+        logger "Stopping services"
+        /Applications/Huntress.app/Contents/MacOS/HuntressUpdater stop
+        /Applications/Huntress.app/Contents/MacOS/HuntressAgent stop
+        /Applications/Huntress.app/Contents/MacOS/Huntress daemonctl stop
+
+        logger "Running updater uninstall..."
+        /Applications/Huntress.app/Contents/MacOS/HuntressUpdater uninstall
+
+        logger "Running agent uninstall..."
+        /Applications/Huntress.app/Contents/MacOS/HuntressAgent uninstall
+
+        logger "Uninstalling GRPC daemon..."
+        /Applications/Huntress.app/Contents/MacOS/Huntress daemonctl uninstall
+
+        logger "Removing app directory..."
+        rm -rf /Applications/Huntress.app
+
+        logger "Removing support directory..."
+        rm -rf "/Library/Application Support/Huntress"
+    else 
+        logger "Huntress assets not found on disk, no services to stop and no assets to delete. Script continuing..."
+    fi
+
+    pkgCheck=$(checkForPkg)     
+    if [ $pkgCheck == 1 ]; then
+        logger "Forgetting package id..."
+        pkgutil --forget com.huntresslabs.pkg.agent 2>/dev/null
+    else 
+        logger "Huntress pkg not found. Script continuing..."
+    fi
+
+    # wait timer for all components to finish uninstall before proceeding with install
+    sleep 3
+    logger "Uninstall complete, beginning reinstall"
+}
+
+# Uninstall the Huntress system extension
+uninstall_system_extension() {
+    logger "Uninstalling system extension"
+    tmp_plist=/tmp/com.apple.system-extensions.admin.plist
+    tmp_plist_prev=/tmp/com.apple.system-extensions.admin.prev.plist
+
+    logger "Toggling system authorization settings..."
+    security authorizationdb read com.apple.system-extensions.admin > $tmp_plist 
+    cp $tmp_plist $tmp_plist_prev
+
+    /usr/libexec/PlistBuddy -c "Set rule:0 is-root" $tmp_plist
+    security authorizationdb write com.apple.system-extensions.admin < $tmp_plist
```

Comment:
I'm not familiar with this setting, what is the reason for doing this?

Reply:
That's to ensure we can uninstall the extension without having to provide a password. It's taken from [this page](https://derflounder.wordpress.com/2021/10/26/silently-uninstalling-system-extensions-on-macos-monterey-and-earlier/) (and, more immediately, from our uninstall script).

Reply:
This does bring up the question of whether we want to consolidate our existing scripts in some way. If we ever need to change the uninstall routine in the future, it would be very easy for us to forget to make the corresponding change in this file.

Reply:
Thanks for the clarification. Yeah, it seems we're getting more spread out on all these different locations for scripts et al.

Reply:
I pulled this from our Uninstall script in \Applications\Huntress\Contents\Scripts\  

I'm not familiar with the command. I tried altering it slightly to fit the script better but that broke it so I went back to the default found in our Scripts folder.